Knoema.Client/Client.cs

```diff
@@ -117,12 +117,18 @@ private Task<T> ApiPost<T>(string path, object obj)
 
 		public Task<IEnumerable<Dataset>> ListDatasets(string source = null, string topic = null, string region = null)
 		{
-			var query = string.Empty
-				.AddUrlParam("source", source)
-				.AddUrlParam("topic", topic)
-				.AddUrlParam("region", region);
 
-			return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset", query);
+			if (string.IsNullOrEmpty(source ?? topic ?? region))
+				return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
+			else
+			{
+				var payload = new Dictionary<string, string>();
```

Comment:
we have this code:

On Fri, Aug 21, 2015 at 5:40 PM, rpanday <notifications@github.com> wrote:

> In Knoema.Client/Client.cs
> <https://github.com/Knoema/knoema-csharp-driver/pull/15#discussion_r37629964>
> :
>
> >
> > -			return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset", query);
> > +			if (string.IsNullOrEmpty(source ?? topic ?? region))
> > +				return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
> > +			else
> > +			{
> > +				var payload = new Dictionary<string, string>();
>
> Simple fix will cause the same problem in communities with no
> customMetadataFields
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Knoema/knoema-csharp-driver/pull/15/files#r37629964>.
>



-- 
Evgeniya Polyakova,
Knoema